cmd/exporter/exporter.go

```diff
@@ -1,36 +1,58 @@
 package exporter
 
 import (
-	"log"
 	"net/http"
 	"os"
 	"os/signal"
 	"syscall"
 
 	"github.com/app-sre/acs-cve-prom-exporter/internal/acs"
+	"github.com/go-kit/log"
+	"github.com/go-kit/log/level"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-func Run(endpoint string, token string) int {
+type RunParams struct {
+	Endpoint      string
+	Token         string
+	ListenAddress *string
+	MetricsPath   *string
+	Logger        log.Logger
+}
+
+func Run(params RunParams) int {
 	reg := prometheus.NewRegistry()
 	exit := make(chan struct{})
 	term := make(chan os.Signal, 1)
 	signal.Notify(term, os.Interrupt, syscall.SIGTERM)
-	go acs.Run(reg, endpoint, token, exit)
-
+	credentials := acs.ACSCredentials{
+		Endpoint: params.Endpoint,
+		Token:    params.Token,
+	}
+	go acs.Run(reg, params.Logger, credentials, exit)
+	http.Handle(*params.MetricsPath, promhttp.HandlerFor(reg, promhttp.HandlerOpts{}))
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`<html>
+             <head><title>ACS CVE Exporter</title></head>
+             <body>
+             <h1>ACS CVE Exporter</h1>
+             <p><a href='` + *params.MetricsPath + `'>Metrics</a></p>
+             </body>
+             </html>`))
+	})
```

Comment:
is this being used as health check for probes?

Reply:
@hemslo  it is a part of best pracrtices for writing exporter (https://prometheus.io/docs/instrumenting/writing_exporters/#landing-page) . But can be leveraged for health checks

Reply:
nice, let's make sure we have health checks setup for deployment, in the future updates.